### PR TITLE
Add functionality to display the correct information on check details screen

### PR DESCRIPTION
**Pull request by caismith-ch**
Resolves: GCI-552

### CODE REVIEW

File: src/controllers/check.details.controller.ts [hunk 1]
```diff
@@ -20,4 +51,69 @@ const route = async (req: Request, res: Response, next: NextFunction) => {
     }
 };
 
+function mapIncludedOnCertificate(itemOptions: ItemOptions): string {
+
+    let mappings = new Array<string>();
```
Comment (nabeelamirch) [inline src/controllers/check.details.controller.ts:56]: Don't forget the other fields
```
const mapIncludedOnCertificate = (itemOptions: Record<string, any>): string => {
    const includedOnCertificate: string[] = [];
    if (itemOptions?.directorDetails?.includeBasicInformation === true) {
        includedOnCertificate.push("Directors");
    }
    if (itemOptions?.includeCompanyObjectsInformation === true) {
        includedOnCertificate.push("Company objects");
    }
    if (itemOptions?.includeGoodStandingInformation === true) {
        includedOnCertificate.push("Statement of good standing");
    }
    if (itemOptions?.registeredOfficeAddressDetails?.includeAddressRecordsType?.length !== 0) {
        includedOnCertificate.push("Registered office address");
    }
    if (itemOptions?.secretaryDetails?.includeBasicInformation === true) {
        includedOnCertificate.push("Secretaries");
    }
    return includedOnCertificate.reduce((accum, value) => accum + "<br>" + value);
};
```

File: src/controllers/check.details.controller.ts [hunk 2]
```diff
@@ -20,4 +51,69 @@ const route = async (req: Request, res: Response, next: NextFunction) => {
...
+
+    mappings.forEach(element => {
+        htmlString += element + "<br>";
+    });
+    return htmlString;
+}
+
+function mapCertificateType(certificateType: string): string {
+    const typeCapitalised = certificateType.charAt(0).toUpperCase()
+    + certificateType.slice(1);
```
Comment (nabeelamirch) [inline src/controllers/check.details.controller.ts:110]: I think this is right, but just to be sure _incorporation-with-all-name-changes_ should be _Incorporation with all name changes_

File: src/controllers/check.details.controller.ts [hunk 3]
```diff
@@ -2,8 +2,39 @@ import { getAccessToken } from "../session/helper";
...
+        return res.render(CHECK_DETAILS, {
+            companyName: certificateItem.companyName,
+            companyNumber: certificateItem.companyNumber,
+            certificateType: mapCertificateType(itemOptions.certificateType),
+            fee: applyCurrencySymbol(certificateItem.itemCosts[0].itemCost),
+            certificateMappings: mapIncludedOnCertificate(itemOptions),
+            changeIncludedOn: replaceCertificateId(CERTIFICATE_OPTIONS, req.params.certificateId),
+            changedeliveryDetails: replaceCertificateId(DELIVERY_DETAILS, req.params.certificateId),
+            deliveryDetails: mapDeliveryDetails(basket.deliveryDetails),
+            templateName: CHECK_DETAILS,
```
Comment (nabeelamirch) [inline src/controllers/check.details.controller.ts:30]: You could add the delivery timescale:
```
const mapDeliveryMethod = (itemOptions: Record<string, any>): string | null => {
    if (itemOptions?.deliveryTimescale === "standard") {
        return "Standard delivery (dispatched within 4 working days)";
    }
    if (itemOptions?.deliveryTimescale === "same-day") {
        return "Same Day";
    }
    return null;
};
```
But it is always going to be _Standard delivery (dispatched within 4 working days)_ for mvp

File: src/controllers/check.details.controller.ts [hunk 4]
```diff
@@ -20,4 +51,69 @@ const route = async (req: Request, res: Response, next: NextFunction) => {
...
+    }
+
+    mappings.push(deliveryDetails.country);
+
+    return mapToHtml(mappings);
+}
+
+function mapToHtml(mappings: Array<string>): string {
+    let htmlString: string = "";
+
```
Comment (nabeelamirch) [inline src/controllers/check.details.controller.ts:101]: I might use this in my code!

File: src/controllers/check.details.controller.ts [hunk 5]
```diff
@@ -20,4 +51,69 @@ const route = async (req: Request, res: Response, next: NextFunction) => {
...
+    }
+
+    if (itemOptions.includeCompanyObjectsInformation) {
+        mappings.push(COMPANY_OBJECTS);
+    }
+
+    return mapToHtml(mappings);
+}
+
+function mapDeliveryDetails(deliveryDetails: DeliveryDetails | undefined): string {
```
Comment (nabeelamirch) [inline src/controllers/check.details.controller.ts:69]: I would change these to fat arrow functions. Also could you add unit tests for these in a file called check.details.controller.spec.unit.ts

File: src/controllers/check.details.controller.ts (hunk 1 above)
Comment (caismith-ch) [reply src/controllers/check.details.controller.ts:60]: Added the missing items.

File: src/controllers/check.details.controller.ts (hunk 3 above)
Comment (caismith-ch) [reply src/controllers/check.details.controller.ts:34]: Added

File: src/controllers/check.details.controller.ts (hunk 5 above)
Comment (caismith-ch) [reply src/controllers/check.details.controller.ts]: 👍

File: src/controllers/check.details.controller.ts (hunk 2 above)
Comment (nabeelamirch) [reply src/controllers/check.details.controller.ts:136]: You probably know this already, but this has changed again. It should be "Incorporation with all company name changes". I did the following in my code:
if (cerificateType === "incorporation-with-all-name-changes") {
        return "Incorporation with all company name changes";
    }

Comment (caismith-ch) [reply src/controllers/check.details.controller.ts:141]: Just update as you added the comment

**Pull request merged**